src/pcm_lowlevel_u585ai.cpp

```diff
@@ -0,0 +1,407 @@
+  DMic_OnBoard_AdfFilterConfig.SnapshotFormat                     = MDF_SNAPSHOT_23BITS;
+  DMic_OnBoard_AdfFilterConfig.Trigger.Source                     = MDF_FILTER_TRIG_TRGO;
+  DMic_OnBoard_AdfFilterConfig.Trigger.Edge                       = MDF_FILTER_TRIG_RISING_EDGE;
+  DMic_OnBoard_AdfFilterConfig.AcquisitionMode                    = MDF_MODE_SYNC_CONT;
+
+  return ret;
+}
+
+int pcm_lowlevel_deinit()
+{
+  int ret = PCM_OK;
+
+  /* De-initializes DFSDM Filter handle */
+  if (DMic_OnBoard_AdfHandle.Instance != NULL) {
+    if (HAL_MDF_DeInit(&DMic_OnBoard_AdfHandle) != HAL_OK) {
+      ret =  PCM_ERROR;
+    }
+    DMic_OnBoard_AdfHandle.Instance = NULL;
+  }
+
+  return ret;
+}
+
+int pcm_lowlevel_start()
+{
+  int ret = PCM_OK;
+
+  /* Initialize DMA configuration parameters */
+  DMic_OnBoard_DMAConfig.Address    = (uint32_t)&DMic_OnBoard_Buffer[0];
+  DMic_OnBoard_DMAConfig.DataLength = (DEFAULT_AUDIO_IN_BUFFER_SIZE * 2U);
+  DMic_OnBoard_DMAConfig.MsbOnly    = ENABLE;
+
+  if (HAL_MDF_AcqStart_DMA(
+        &DMic_OnBoard_AdfHandle,
+        &DMic_OnBoard_AdfFilterConfig,
+        &DMic_OnBoard_DMAConfig) != HAL_OK) {
+    ret = PCM_ERROR;
+  }
+
+  if (HAL_MDF_GenerateTrgo(&DMic_OnBoard_AdfHandle) != HAL_OK) {
+    ret = PCM_ERROR;
+  }
+
+  return ret;
+}
+
+int pcm_lowlevel_stop()
+{
+  int ret = PCM_OK;
+
+  if (HAL_MDF_AcqStop_DMA(&DMic_OnBoard_AdfHandle) != HAL_OK) {
+    ret = PCM_ERROR;
+  }
+
+  return PCM_OK;
+}
+
+/**
+* @brief  Regular conversion complete callback.
+* @note   In interrupt mode, user has to read conversion value in this function
+using HAL_DFSDM_FilterGetRegularValue.
+* @param  hmdf   MDF filter handle.
+* @retval None
+*/
+void HAL_MDF_AcqCpltCallback(MDF_HandleTypeDef *hmdf)
+{
+  UNUSED(hmdf);
+
+  if (PCM.Buffer == nullptr) {
+    return;
+  }
+
+  if (DMic_OnBoard_CountSkip < 64) {
+    ++DMic_OnBoard_CountSkip;
+    return;
+  }
+
+  for (uint32_t j = 0U; j < ((PCM.SampleRate / 1000U) * N_MS_PER_INTERRUPT); j++) {
+    int32_t Z = ((DMic_OnBoard_Buffer[j + ((PCM.SampleRate / 1000U) * N_MS_PER_INTERRUPT)]) * (int32_t)(PCM.Volume)) / 100;
+    PCM.Buffer[j] = (uint16_t) SaturaLH(Z, -32760, 32760);
+  }
+
+  if (PCM.OnRecv != nullptr) {
+    PCM.OnRecv();
+  }
+}
+
+/**
+* @brief  Half regular conversion complete callback.
+* @param  hmdf   MDF filter handle.
+* @retval None
+*/
+void HAL_MDF_AcqHalfCpltCallback(MDF_HandleTypeDef *hmdf)
+{
+  UNUSED(hmdf);
+
+  if (PCM.Buffer == nullptr) {
+    return;
+  }
+
+  if (DMic_OnBoard_CountSkip < 64) {
+    ++DMic_OnBoard_CountSkip;
+    return;
+  }
+
+  for (uint32_t j = 0U; j < ((PCM.SampleRate / 1000U) * N_MS_PER_INTERRUPT); j++) {
+    int32_t Z = (DMic_OnBoard_Buffer[j] * (int32_t)(PCM.Volume)) / 100;
+    PCM.Buffer[j] = (uint16_t) SaturaLH(Z, -32760, 32760);
+  }
+
+  if (PCM.OnRecv != nullptr) {
+    PCM.OnRecv();
+  }
+}
+
+/**
+  * @brief System Clock Configuration
+  * @retval None
+  */
+void SystemClock_Config(void)
```

Comment:
I guess this should be removed and update the default one in the variant.

Reply:
As far as I can tell you, this clock configuration is needed only when using this library.

Reply:
Hi @grobx 
main issue here is you redefine the `SystemClock_Config()` which prevents user to redefine it on its own.
So the weak  `SystemClock_Config()` available in the variant should be updated to add the required clock config you need.

Reply:
Question: is the default  SystemClock_Config defined in the variant is correct and allow you to have this library working else we need to update it in the core?

Reply:
Fine.